mlrun/frameworks/_ml_common/pkl_model_server.py

```diff
@@ -36,11 +37,15 @@ def predict(self, body: dict) -> list:
         Infer the inputs through the model using MLRun's PyTorch interface and return its output. The inferred data will
         be read from the "inputs" key of the request.
 
-        :param request: The request of the model. The input to the model will be read from the "inputs" key.
+        :param body: The request of the model. The input to the model will be read from the "inputs" key.
 
         :return: The model's prediction on the given input.
         """
-        x = np.asarray(body["inputs"])
+        inputs = body["inputs"]
+        if inputs and isinstance(inputs[0], dict):
+            x = pd.DataFrame(inputs[0])
```

Comment:
can you explain why do you ignore the rest of the inputs list?

Reply:
When passing a dict in `request["inputs"]` the data is divided into features, meaning, the keys are the feature names and each value contains a list that with all the data of that feature. So we look only on the first element because it contains all the data in this case.